neutrino.go

```diff
@@ -666,6 +666,7 @@ type ChainService struct { // nolint:maligned
 	broadcaster          *pushtx.Broadcaster
 	banStore             banman.Store
 	workManager          *query.WorkManager
+	queryDispatcher      query.Dispatcher
```

Comment:
and cant switch out the current `*query.WorkManager` for the `query.Disptcher` interface since the interface doesnt have the `Start()` and `Stop()` methods.... although - I could just add those to the interface? Lemme know what you think

Reply:
cool I added a commit that hopefully makes this nicer. Lemme know what you think :)